puppet/zulip/files/nginx/zulip-include-frontend/app

```diff
@@ -49,6 +49,26 @@ location ~ ^/tornado/(\d+)(/.*)$ {
     include /etc/nginx/zulip-include/proxy_longpolling;
 }
 
+# Handle redirects to S3
+location ~ ^/user_uploads/s3/([^/]+)/(.*) {
+    internal;
+    set $download_url https://$1/$2;
+    proxy_set_header Host $1;
+
+    # Inherit the Content-Disposition which Django sent
+    proxy_hide_header Content-Disposition;
+    add_header Content-Disposition $upstream_http_content_disposition;
+    include /etc/nginx/zulip-include/headers;
+
+    resolver 127.0.0.53 valid=300s;
+    resolver_timeout 10s;
```

Comment:
Why do we need a resolver here, and why is that IP correct? Maybe worth a comment.